lib/logstash/inputs/unix.rb

```diff
@@ -140,23 +138,26 @@ def run(output_queue)
             raise
           end
         end
-      end # loop
+      end
     else
-      loop do
-        client_socket = UNIXSocket.new(@path)
-        client_socket.instance_eval { class << self; include ::LogStash::Util::SocketPeer end }
+      while !stop?
+        @client_socket = UNIXSocket.new(@path)
+        @client_socket.instance_eval { class << self; include ::LogStash::Util::SocketPeer end }
         @logger.debug("Opened connection", :client => @path)
-        handle_socket(client_socket, output_queue)
-      end # loop
+        handle_socket(@client_socket, output_queue)
+      end
     end
   end # def run
 
   public
-  def teardown
+  def stop
+    super
     if server?
       File.unlink(@path)
```

Comment:
This is actually wrong, this variable is also used in server mode, see the code https://github.com/logstash-plugins/logstash-input-unix/blob/master/lib/logstash/inputs/unix.rb#L47

is unlinked here for whatever reason who implemented this thought, but kinda make sense as server sockets might be only working when the server is up, then creating a new one after all.

Client sockets should never delete the socket as is not their responsibility, they just connect to it to read.